[fix bug 1293779] Add homepage Bv2 variations.

jpetto opened the pull request:
## Description

Make old homepage variant B the new default for en-US visitors. Add 3 variations of B for testing:

1. Add slideshow from old variant A (and a few other UI tweaks)
2. Update copy
3. Add slideshow & update copy

Also removes old variant A.

## Bugzilla link

https://bugzilla.mozilla.org/show_bug.cgi?id=1293779

## Testing

Ensure pages are loading and functioning as expected:

- `/en-US/`
- `/en-US/?v=a`
- `/en-US/?v=b`
- `/en-US/?v=c

## Checklist
- [ ] Related functional & integration tests passing.

jpetto commented:
Marked as do not merge for now as we are still waiting on an updated survey link.

alexgibson commented on bedrock/mozorg/templates/mozorg/home/home.html:
```diff
@@ -35,12 +35,6 @@
   <![endif]-->
 {% endblock %}
 
-{% block optimizely %}
```
Not opposed to removing this, just making sure this is intentional?

jpetto replied:
Yeah, since this new homepage is for en-US, and we pretty much only test on en-US, I figured it's safe to remove from this (now) non en-US template.

alexgibson commented on bedrock/mozorg/templates/mozorg/home/home-en-US-a.html:
```diff
@@ -0,0 +1,38 @@
+
+{% block js %}
+  {{ super() }}
+  {% javascript 'home-en-US-a' %}
```
I don't seem to see this bundle name in static media?

jpetto replied:
Gah! Originally I had a separate JS bundle, but refactored into one. I'm going to ask pmac about configuring bedrock (locally, at least) to throw an error when a bundle doesn't exist.

alexgibson commented on bedrock/mozorg/templates/mozorg/home/home-en-US-c.html:
```diff
@@ -0,0 +1,51 @@
+
+{% block js %}
+  {{ super() }}
+  {% javascript 'home-en-US-a' %}
```
Same as above

alexgibson commented on bedrock/mozorg/views.py:
```diff
@@ -254,10 +254,14 @@ def home_tweets(locale):
+        variant = request.GET.get('v', '')
+
+        if variant in ['a', 'b', 'c']:
+            template = 'mozorg/home/home-en-US-{0}.html'.format(variant)
```
Not a blocker, but it would be nice to have some tests for the view logic?

jpetto replied:
Agreed. Will add.

alexgibson commented on media/js/mozorg/home/survey.js:
```diff
@@ -13,7 +13,7 @@ if (typeof Mozilla === 'undefined') {
         copyIntro: 'Hello! Would you be willing to take a minute to answer a few questions for Mozilla?',
         copyLink: 'Sure. I\'ll help.',
-        surveyURL: 'https://www.surveygizmo.com/s3/2841272/Brand-Perception-Survey-2016-Homepage-Prototype-A-bottom',
+        surveyURL: 'NEW SURVEY LINK NEEDED',
```
Just noting the TODO here 👍

alexgibson commented on bedrock/mozorg/templates/mozorg/home/home-en-US.html:
```diff
@@ -239,8 +272,9 @@ <h1 class="logo"><a href="{{ url('mozorg.home') }}" data-link-type="footer" data
 
-  {# survey is en-US only - enforced via optimizely audience segmentation #}
-  {% include 'mozorg/home/includes/survey-b.html' %}
+  {% if LANG == 'en-US' %}
```
Is this conditional needed?

jpetto replied:
Oh, yeah, nope, not anymore. Will remove.

alexgibson commented:
Tested this in modern browsers, plus IE9 and things look good. I think most of what is here has already been reviewed in some past iteration or another. Aside from the small nits / suggestions, this gets an r+wc from me 👍

jpetto commented:
@alexgibson - Addressed review suggestions, and made a slight change to the docs after chatting w/pmac. (`requirements/test.txt` is required to run our tests, and it installs `requirements/dev.txt` as a dependency, so...yeah.)

alexgibson commented:
Changes look good to me, r+ once squashed and you're ready to merge (survey link pending) ⚡

jpetto commented:
@alexgibson - Squashed previous commits, but added another for survey links. (I didn't realize each variation would have a unique link, hence the extra commit for easier review.) If this last commit is good, I'll squash once more.

Thanks!

alexgibson commented:
This looks good to me, r+ to merge when ready!
